src/test/java/org/mifos/integrationtest/cucumber/stepdef/VoucherManagementStepDef.java

```diff
@@ -295,12 +306,10 @@ public void createAndActivateVoucher() {
         mockServerStepDef.startStub("/createVoucher", PUT, 200);
         mockServerStepDef.startStub("/activateVoucher", PUT, 200);
         iCallTheVoucherCreateAPIWithExpectedStatusOf(202, "/createVoucher");
-        try {
-            Thread.sleep(5000);
-        } catch (InterruptedException e) {
-            throw new RuntimeException(e);
-        }
-        iShouldBeAbleToExtractResponseBodyFromCallback();
+        await().atMost(20, SECONDS).pollInterval(1, MILLISECONDS).until(() -> {
+            iShouldBeAbleToExtractResponseBodyFromCallback();
+            return true; // Replace this with the actual condition you are waiting for
```

Comment:
This needs change.
The comment is a todo. (Replace this with the actual condition you are waiting for)